File: fix_jsx_semicolons.py
#!/usr/bin/env python3
"""
Fix semicolon issues in JSX files.
"""

import os
import re
from pathlib import Path

def fix_jsx_semicolons(content):
    """Fix semicolons in JSX content."""
    # Fix semicolons in JSX text content
    content = re.sub(r'([^;])\s*;\s*</', r'\1</', content)
    content = re.sub(r'([^;])\s*;\s*<br', r'\1<br', content)
    content = re.sub(r'([^;])\s*;\s*<span', r'\1<span', content)
    content = re.sub(r'([^;])\s*;\s*<div', r'\1<div', content)
    content = re.sub(r'([^;])\s*;\s*<h[1-6]', r'\1<h', content)
    content = re.sub(r'([^;])\s*;\s*<p', r'\1<p', content)
    
    # Fix semicolons in JSX attributes
    content = re.sub(r'className="([^"]*);"', r'className="\1"', content)
    content = re.sub(r'id="([^"]*);"', r'id="\1"', content)
    
    return content

def process_file(file_path):
    """Process a single file to fix JSX semicolons."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original_content = content
        content = fix_jsx_semicolons(content)
        
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            print(f"Fixed: {file_path}")
            return True
        else:
            print(f"No changes needed: {file_path}")
            return False
            
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to process all files."""
    workspace = Path('/workspace')
    
    # File patterns to process
    patterns = [
        'app/**/*.tsx',
        'app/**/*.jsx',
        'src/**/*.tsx',
        'src/**/*.jsx',
    ]
    
    total_files = 0
    fixed_files = 0
    
    for pattern in patterns:
        for file_path in workspace.glob(pattern):
            if file_path.is_file():
                total_files += 1
                if process_file(file_path):
                    fixed_files += 1
    
    print(f"\nProcessed {total_files} files, fixed {fixed_files} files")

if __name__ == "__main__":
    main()

File: fix-jsx-semicolons.js
const fs = require('fs");"'"
const path = require('path");

// Function to fix JSX semicolon issues;
function fixJSXSemicolons(filePath) {"
;"
try{;"'"
let content = fs.readFileSync(filePath, 'utf8");
    let modified = false;
"
    // Fix semicolons in JSX where there should be commas;"
,}"'"
content = content.replace(/<\/div>;\s*\);/g, '</div>\n  );\n}");"'"
    content = content.replace(/<\/>;\s*\);/g, '</>\n  );\n}");"'"
    content = content.replace(/<\/div>;\s*$/gm, '</div>");"'"
    content = content.replace(/<\/>;\s*$/gm, '</>");"
"
    // Fix malformed JSX structure;"'"
content = content.replace(/(\s*)<div />\s*$/gm, '$1  <div />");"'"
    content = content.replace(/(\s*)<\/div>\s*$/gm, '$1  </div>");"
"
    // Fix missing closing braces;"'"
if (!content.trim().endsWith('}")) {;"'"
content = content.trim() + '\n,}";
      modified = true;
    ,}"
;"
if (modified) {;"'"
fs.writeFileSync(filePath, content, 'utf8");
      console.log(`Fixed JSX semicolons in ${filePath;}`);
      return true;
    }
    return false;
  } catch (error) {;
console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Function to find and fix all page files;"
function fixAllPageFiles() { "
;"'"
const appDir = './app";
  const files = fs.readdirSync(appDir);
  let fixedCount = 0;
;"
files.forEach(file = > {;"
;)"'"
const filePath = path.join(appDir, file, 'page.tsx");
    if (fs.existsSync(filePath)) {;
if (fixJSXSemicolons(filePath)) {;
fixedCount++;
      

, }
    }

    // Also check subdirectories;
const subDir = path.join(appDir, file);
    if (fs.statSync(subDir).isDirectory()) {;
const subFiles = fs.readdirSync(subDir);"
      subFiles.forEach(subFile = > {;"
;)"'"
if(subFile === 'page.tsx") {  ;
const subFilePath = path.join(subDir, subFile);
          if (fixJSXSemicolons(subFilePath)) {;
fixedCount++;
          
, , }
        }
      });
    }
  });
;
return fixedCount;
}"
"
// Main execution;"'"
console.log('Starting JSX semicolon fixes...");"
const fixedCount = fixAllPageFiles();"
console.log(`Fixed ${fixedCount,} files with JSX semicolon issues.`);"'"
console.log('JSX semicolon fixes completed.");"'"